### PR TITLE
Fix 5.67 regression on loading custom data defaults on contribution page

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes a 5.67 regression where we stopped loading the default value for non-membership custom fields

https://lab.civicrm.org/dev/core/-/issues/4230

Before
----------------------------------------
The code to set profile defaults was not being called for non-membership custom fields

After
----------------------------------------
default values load based on the relevant entity or the field defaults if no relevant entity

Technical Details
----------------------------------------
in 5.67 https://lab.civicrm.org/dev/core/-/issues/4230 was fixed with https://github.com/civicrm/civicrm-core/pull/26047, addressing an issue where the membership ID was used to load contribution ID defaults - but it did it blocking the loading of those defaults for custom fields that were non-membership. For logged in users the defaults were loaded earlier in the function but for non-logged in users there was no other place they were being loaded 

This makes it so the custom field defaults for all custom fields are loaded at the start, and potentially overridden further down by those relevant to the contact or membership

Comments
----------------------------------------

### CODE REVIEW

**Comment by pradpnayak** [comment]
Hi @eileenmcnaughton 

Thanks for fixing the issue, I have tested this PR and it solves my original reported issue.

File: CRM/Contribute/Form/Contribution/Main.php [hunk 1]
```diff
@@ -1898,4 +1889,47 @@ private function isPageHasPaymentProcessorSupportForRecurring(): bool {
...
+      }
+      elseif (array_key_exists($name, $contribFields) || (substr($name, 0, 11) === 'membership_') || (substr($name, 0, 13) == 'contribution_')) {
+        continue;
+      }
+      $fields[$name] = $fieldInfo;
+    }
+    return $fields;
+  }
+
+  private function getProfileCustomFields (): array {
```
Comment (pradpnayak) [inline CRM/Contribute/Form/Contribution/Main.php:1921]: Doc block?

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution/Main.php:1933]: added!

**Comment by eileenmcnaughton** [comment]
thanks @pradpnayak  - did you test if it fixed the regression too - or just that it didn't regress the thing you fixed?

**Comment by totten** [comment]
civibot, test this please

**Comment by colemanw** [comment]
retest this please

**Pull request merged**